tests/cupy_tests/creation_tests/test_ranges.py

```diff
@@ -205,3 +210,73 @@ def test_meshgrid3(self, xp, dtype):
         y = xp.arange(3).astype(dtype)
         z = xp.arange(4).astype(dtype)
         return xp.meshgrid(x, y, z, indexing=self.indexing, copy=self.copy)
+
+
+@testing.gpu
+class TestMgrid(unittest.TestCase):
+
+    @testing.for_all_dtypes()
+    def test_mgrid0(self, dtype):
+        out = cupy.mgrid[0:]
+        self.assertListEqual(out, [])
```

Comment:
This assertion fails because `out` is `cupy.ndarray`.
How about testing with `testing.numpy_cupy_array_equal`?